third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc

```diff
@@ -1429,56 +1432,129 @@ ProtocolObjectPreview.prototype = {
     Array_push.call(this.containerEntries, entry);
   },
 
-  get pageIndex() { return 0; },
-
-  get pageSize() {
-    if (isBlinkObject(this.raw, this.cdpObj) || CustomPreviewers[this.cdpObj.className]) {
```

Comment:
1. Yes you are right. I think I added the `|| CustomPreviewers...` check a while back basically exactly because of the `pageSize` issues that now require us to loop. However that is a big performance hit, because big `Array`, `Set`, `Map` etc. are now not paged anymore. So now I thought, let's finally remove it again!
2. However... There is a concern with `Set` and `Map`, because they store their data in `internalProperties`, which we are currently not looping.
3. I thus added a check for a **_small_** `Set` [here](https://github.com/replayio/backend/pull/9313/files#diff-9bc33484e9051eb27d451a451dfb9b2094d530c8a1275d5ff8c612b9c8286065R130). I saw it pass and moved on. But you make the right point: Let's make sure this works quite right.

I'll add more tests and fix more bugs, as I should have from the beginning!